pkg/ipamd/ipamd.go

```diff
@@ -2069,12 +2069,31 @@ func (c *IPAMContext) AnnotatePod(podName, podNamespace, key, val string) error
 		if newPod.Annotations == nil {
 			newPod.Annotations = make(map[string]string)
 		}
-		newPod.Annotations[key] = val
+		// On CNI ADD, always set new annotation
+		oldVal, ok := newPod.Annotations[key]
+		if newVal != "" {
+			//If the new pod annotation is as the same as the old one, return immediately without extra cost
```

Comment:
nit: space after comment, and maybe rephrase it as "Skip patch operation if new value is the same as existing value"